themis/spec/asymmetric-keypairs/rsa.md

```diff
@@ -0,0 +1,445 @@
+jvUf8FJz3XSkN9AYyHLbXSDJqKZnLPJdAAEAAQ==`
+	privateKey, _ := base64.StdEncoding.DecodeString(privateKeyBase64)
+
+	// Extract the values of key parameters
+	d := new(big.Int).SetBytes(privateKey[0x00C:0x10C])
+	p := new(big.Int).SetBytes(privateKey[0x10C:0x18C])
+	q := new(big.Int).SetBytes(privateKey[0x18C:0x20C])
+	dp := new(big.Int).SetBytes(privateKey[0x20C:0x28C])
+	dq := new(big.Int).SetBytes(privateKey[0x28C:0x30C])
+	qp := new(big.Int).SetBytes(privateKey[0x30C:0x38C])
+	n := new(big.Int).SetBytes(privateKey[0x38C:0x48C])
+	e := new(big.Int).SetBytes(privateKey[0x48C:0x490])
+
+	// p and q must be primes (well, with a 1 − 1/4^25 confidence)
+	if p.ProbablyPrime(25) && q.ProbablyPrime(25) {
+		fmt.Println("✅ p and q are primes")
+	} else {
+		fmt.Println("❌ p and q are primes")
+	}
+	// ...and by definition, n = p⋅q
+	nComputed := new(big.Int).Mul(p, q)
+	if nComputed.Cmp(n) == 0 {
+		fmt.Println("✅ n = p⋅q")
+	} else {
+		fmt.Println("❌ n = p⋅q")
+	}
+
+	pMinusOne := new(big.Int).Sub(p, big.NewInt(1))
+	qMinusOne := new(big.Int).Sub(q, big.NewInt(1))
+
+	// By definition, λ(n) = λ(p⋅q) = LCM(λ(p), λ(q)) =
+	//   = LCM(p−1, q−1) = (p−1)(q−1) / GCD(p−1, q−1)
+	pq1Mul := new(big.Int).Mul(pMinusOne, qMinusOne)
+	pq1GCD := new(big.Int).GCD(nil, nil, pMinusOne, qMinusOne)
+	totient := new(big.Int).Div(pq1Mul, pq1GCD)
+
+	// e must be within 1 < e < λ(n)
+	if (big.NewInt(1).Cmp(e) == -1) && (e.Cmp(totient) == -1) {
+		fmt.Println("✅ 1 < e < λ(n)")
+	} else {
+		fmt.Println("❌ 1 < e < λ(n)")
+	}
+	// ...plus, e and λ(n) must be coprime (i.e., GCD(e, λ(n)) = 1)
+	gcd := new(big.Int).GCD(nil, nil, e, totient)
+	if gcd.Cmp(big.NewInt(1)) == 0 {
+		fmt.Println("✅ e is coprime with λ(n)")
+	} else {
+		fmt.Println("❌ e is coprime with λ(n)")
+	}
+
+	// It must be that d⋅e ≡ 1 (mod λ(n))
+	dXe := new(big.Int).Mod(new(big.Int).Mul(d, e), totient)
+	if dXe.Cmp(big.NewInt(1)) == 0 {
+		fmt.Println("✅ d⋅e ≡ 1 (mod λ(n))")
+	} else {
+		fmt.Println("❌ d⋅e ≡ 1 (mod λ(n))")
+	}
+
+	// By definition, dp ≡ d (mod p−1)
+	dpComputed := new(big.Int).Mod(d, pMinusOne)
+	if dpComputed.Cmp(dp) == 0 {
+		fmt.Println("✅ dp ≡ d (mod p−1)")
+	} else {
+		fmt.Println("❌ dp ≡ d (mod p−1)")
+	}
+
+	// By definition, dq ≡ d (mod q−1)
+	dqComputed := new(big.Int).Mod(d, qMinusOne)
+	if dqComputed.Cmp(dq) == 0 {
+		fmt.Println("✅ dq ≡ d (mod q−1)")
+	} else {
+		fmt.Println("❌ dq ≡ d (mod q−1)")
+	}
+
+	// It must be that qp⋅q ≡ 1 (mod p)
+	qpXq := new(big.Int).Mod(new(big.Int).Mul(qp, q), p)
+	if qpXq.Cmp(big.NewInt(1)) == 0 {
+		fmt.Println("✅ qp⋅q ≡ 1 (mod p)")
+	} else {
+		fmt.Println("❌ qp⋅q ≡ 1 (mod p)")
+	}
+}
+```
+
+## Generating RSA keys
+
+The recommended way to generate RSA keypairs is to use a cryptography library.
+
+See also our [key management guide](/themis/crypto-theory/key-management/)
+for advice on securing the keys after you have generated them.
+
+Here is a snippet in Go, illustrating how to generate a good RSA keypair:
```

Comment:
```suggestion
Here is a snippet in Go, illustrating how to generate a long RSA keypair:
```

Reply:
The point here isn't to have a long keypair. If you're concerned, how about we drop the word completely?
```suggestion
Here is a snippet in Go, illustrating how to generate an RSA keypair:
```

Reply:
I don't think we should be using "good" as description of cryptographically-secure keys/keypairs in official docs, that's why I suggest other options.